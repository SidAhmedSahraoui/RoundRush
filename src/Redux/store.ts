import { Action, ThunkAction, configureStore } from "@reduxjs/toolkit";

import auth from "./auth/auth-slice";


export const store = configureStore({
  reducer: {
    auth
  },
  devTools: true,
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
