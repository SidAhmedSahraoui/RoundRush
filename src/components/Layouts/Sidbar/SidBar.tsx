import React, { Fragment, useState } from "react";
import useStyles from "./style";
import Logo from "../../../img/logos.svg";
import Trello from "../../../img/trello.svg";
import Home from "../../../img/home.svg";
import Settings from "../../../img/settings.svg";
import Member from "../../../img/member.svg";
import Group from "../../../img/Group.svg";
import Oval from "../../../img/oval2.svg";
import Help from "../../../img/help.svg";
import Exit from "../../../img/exit.svg";
import { KeyboardArrowDown, MoreHoriz } from "@mui/icons-material";
import AddIcon from "@mui/icons-material/Add";
import { SvgIcon } from "@mui/material";
import { v4 as uuidv4 } from "uuid";
const SidBar: React.FC = () => {
  const classes = useStyles();
  const [click, setClick] = useState(false);
  const [dispTeam, setDispTeam] = useState(false);
  const [dis, setDis] = useState([false, false, false]);

  const clickHandler = (event: React.MouseEvent<HTMLButtonElement>) => {
    event.preventDefault();
    setClick((click) => !click);
  };

  const displayHandler = (event: React.MouseEvent<SVGSVGElement>) => {
    event.preventDefault();
    setDispTeam((dispTeam) => !dispTeam);
  };
  const displayTeams = (id: number) => {
    const arr = [...dis];
    setDis((dis) => {
      arr[id] = !arr[id];
      return arr;
    });
  };

  const teams = [
    {
      id: 0,
      name: "Coraly",
      sub: [
        {
          name: "Coraly Tech & Dev",
          color: "#EEC301",
        },
        { name: "Design Board", color: "#E7374F" },
        { name: "Growth Hacking", color: "#FC91AD" },
      ],
    },
    {
      id: 1,
      name: "Lasting Dynamics",
      sub: [
        {
          name: "Website",
          color: "#EEC301",
        },
        { name: "Growth Hacking", color: "#E7374F" },
        { name: "UI / UX Design", color: "#FC91AD" },
      ],
    },
    {
      id: 2,
      name: "Vetrina Live",
      sub: [
        {
          name: "VL - Tech & Dev",
          color: "#EEC301",
        },
        { name: "VL - Growth Hacking", color: "#E7374F" },
      ],
    },
  ];
  return (
    <>
      {!click ? (
        <>
          <div  className={classes.sidbar}>
            <button onClick={clickHandler} className="humberger">
              <span></span>
              <span></span>
              <span></span>
            </button>
            <div className="elem">
              <img src={Logo} alt="logo" />
            </div>
            <div className="elem">
              {" "}
              <img src={Trello} alt="trello" />
            </div>
            <div className="elem">
              <img src={Home} alt="home" />
            </div>
            <div className="elem">
              <img src={Settings} alt="settings" />
            </div>
            <span className="divider"></span>
            <div className="elem">
              <img src={Member} alt="member" />
            </div>
            <div className="down">
              <div className="img">
                <img src={Oval} alt="oval" />
              </div>
              <div className="img">
                <img src={Help} alt="help" />
              </div>
              <div className="img">
                <img src={Exit} alt="exit" />
              </div>
            </div>
          </div>
        </>
      ) : (
        <>
          <div style={(dis.includes(true)) ? {height: "130vh"} : {height: "100wh"}} className={classes._sidbar}>
            <div className="_humberger">
              <img src={Group} alt="grp" />
              <button onClick={clickHandler} className="_hum">
                <span></span>
                <span></span>
                <span></span>
              </button>
            </div>
            <div className="_elem">
              <img src={Logo} alt="logo" />
              <h1>YellowTech Inc.</h1>
              <SvgIcon
                sx={{ color: "#FFFFFF", paddingLeft: "12px" }}
                component={KeyboardArrowDown}
              />
            </div>
            <div className="_elem">
              {" "}
              <img src={Trello} alt="trello" />
              <h1>Personal Board</h1>
            </div>
            <div className="_elem">
              <img src={Home} alt="home" />
              <h1>Dashboard</h1>
            </div>
            <div className="_elem">
              <img src={Settings} alt="settings" />
              <h1>Space Settings</h1>
            </div>
            <span className="_divider"></span>
            <div className="_elem">
              <img src={Member} alt="member" />
              <h1>Teams</h1>
              <SvgIcon
                sx={{ color: "#FFFFFF", paddingLeft: "34px" }}
                component={AddIcon}
              />
              <SvgIcon
                sx={{ color: "#FFFFFF", paddingLeft: "12px" }}
                component={KeyboardArrowDown}
                onClick={displayHandler}
              />
            </div>
            {dispTeam
              ? teams.map((team) => (
                  <Fragment key={uuidv4()}>
                    <div key={uuidv4()} className="company">
                      <h5>{team.name}</h5>
                      <div className="cmp">
                        <SvgIcon
                          sx={{ color: "#FFFFFF" }}
                          component={MoreHoriz}
                        />
                        <SvgIcon
                          sx={{ color: "#FFFFFF" }}
                          component={KeyboardArrowDown}
                          onClick={() => displayTeams(team.id)}
                        />
                      </div>
                    </div>
                    {dis[team.id]
                      ? team.sub.map((t) => (
                          <div className="team" key={uuidv4()}>
                            <div
                              className="cube"
                              style={{ backgroundColor: t.color }}></div>
                            <h5>{t.name}</h5>
                            <SvgIcon
                              sx={{ color: "#FFFFFF" }}
                              component={MoreHoriz}
                              className="icon"
                            />
                          </div>
                        ))
                      : null}
                  </Fragment>
                ))
              : null}
            <div className="down">
            <div className="down">
              <div className="img">
                <img src={Oval} alt="oval" />
              </div>
              <div className="img">
                <img src={Help} alt="help" />
              </div>
              <div className="img">
                <img src={Exit} alt="exit" />
              </div>
            </div>
            </div>
          </div>
        </>
      )}
    </>
  );
};

export default SidBar;
