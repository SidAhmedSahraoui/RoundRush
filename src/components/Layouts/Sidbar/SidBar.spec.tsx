import React from "react";
import { cleanup, render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../../../redux/store";
import SidBar from "./SidBar";

describe("SidBar", () => {
  afterEach(() => cleanup());

  test("Render SidBar Component", () => {
    const view = render(
      <Provider store={store}>
        <Router>
          <SidBar />
        </Router>
      </Provider>,
    );

    expect(view.container).toMatchSnapshot();
  });  
});