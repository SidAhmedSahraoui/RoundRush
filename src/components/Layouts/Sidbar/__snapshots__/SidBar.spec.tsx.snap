// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidBar Render Navbar component 1`] = `
<div>
  <div
    class="sidbar-0-2-1"
  >
    <div
      class="container"
    >
      <div
        class="up"
      >
        <button
          class="humberger"
        >
          <span />
          <span />
          <span />
        </button>
        <div
          class="elem"
        >
          <img
            alt="logo"
            src="logos.svg"
          />
        </div>
        <a
          class="elem"
          href="/profile"
        >
           
          <img
            alt="trello"
            src="trello.svg"
          />
        </a>
        <a
          class="elem"
          href="/dashboard"
        >
          <img
            alt="home"
            src="home.svg"
          />
        </a>
        <a
          class="elem"
          href="/objectives"
        >
          <img
            alt="settings"
            src="settings.svg"
          />
        </a>
        <span
          class="divider"
        />
        <div
          class="elem"
        >
          <img
            alt="member"
            src="member.svg"
          />
        </div>
      </div>
      <div
        class="down"
      >
        <div
          class="img"
        >
          <img
            alt="oval"
            src="Oval2.svg"
          />
        </div>
        <div
          class="img"
        >
          <img
            alt="help"
            src="help.svg"
          />
        </div>
        <div
          class="img"
        >
          <img
            alt="exit"
            src="exit.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SidBar Render SidBar Component 1`] = `
<div>
  <div
    class="sidbar-0-2-1"
  >
    <div
      class="container"
    >
      <div
        class="up"
      >
        <button
          class="humberger"
        >
          <span />
          <span />
          <span />
        </button>
        <div
          class="elem"
        >
          <img
            alt="logo"
            src="logos.svg"
          />
        </div>
        <a
          class="elem"
          href="/profile"
        >
           
          <img
            alt="trello"
            src="trello.svg"
          />
        </a>
        <a
          class="elem"
          href="/dashboard"
        >
          <img
            alt="home"
            src="home.svg"
          />
        </a>
        <a
          class="elem"
          href="/objectives"
        >
          <img
            alt="settings"
            src="settings.svg"
          />
        </a>
        <span
          class="divider"
        />
        <div
          class="elem"
        >
          <img
            alt="member"
            src="member.svg"
          />
        </div>
      </div>
      <div
        class="down"
      >
        <div
          class="img"
        >
          <img
            alt="oval"
            src="Oval2.svg"
          />
        </div>
        <div
          class="img"
        >
          <img
            alt="help"
            src="help.svg"
          />
        </div>
        <div
          class="img"
        >
          <img
            alt="exit"
            src="exit.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`src/components/layout/navbar Render Navbar component 1`] = `
<div>
  <div
    class="sidbar-0-2-1"
  >
    <div
      class="container"
    >
      <div
        class="up"
      >
        <button
          class="humberger"
        >
          <span />
          <span />
          <span />
        </button>
        <div
          class="elem"
        >
          <img
            alt="logo"
            src="logos.svg"
          />
        </div>
        <a
          class="elem"
          href="/profile"
        >
           
          <img
            alt="trello"
            src="trello.svg"
          />
        </a>
        <a
          class="elem"
          href="/dashboard"
        >
          <img
            alt="home"
            src="home.svg"
          />
        </a>
        <a
          class="elem"
          href="/objectives"
        >
          <img
            alt="settings"
            src="settings.svg"
          />
        </a>
        <span
          class="divider"
        />
        <div
          class="elem"
        >
          <img
            alt="member"
            src="member.svg"
          />
        </div>
      </div>
      <div
        class="down"
      >
        <div
          class="img"
        >
          <img
            alt="oval"
            src="Oval2.svg"
          />
        </div>
        <div
          class="img"
        >
          <img
            alt="help"
            src="help.svg"
          />
        </div>
        <div
          class="img"
        >
          <img
            alt="exit"
            src="exit.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
